pkg/policy/selectorcache.go

```diff
@@ -33,6 +33,11 @@ type CachedSelector interface {
 	// be modified, as it is shared among multiple users.
 	GetSelections() []identity.NumericIdentity
 
+	// GetMetadataLabels returns metadata labels for additional context
+	// surrounding the selector. These are typically the labels associated with
+	// Cilium rules.
+	GetMetadataLabels() labels.LabelArray
```

Comment:
Thought: I notice that the LabelArray type is being printed as is, it might be nice to implement `fmt.Stringer` on this so that we can provide an easier to read representation of the array.

Reply:
It's hard to predict what the desired representation should be because it's an array. There are other `String()` methods on the related label types so it's not obvious if we need yet another `Stringer`.